packages/browser/src/site-apps.ts

```diff
@@ -80,16 +86,21 @@ export class SiteApps {
             },
             groups,
         }
-        return globals
     }
 
     setupSiteApp(loader: SiteAppLoader) {
         const app = this.apps[loader.id]
         const processBufferedEvents = () => {
             if (!app.errored && this._bufferedInvocations.length) {
                 logger.info(`Processing ${this._bufferedInvocations.length} events for site app with id ${loader.id}`)
-                this._bufferedInvocations.forEach((globals) => app.processEvent?.(globals))
-                app.processedBuffer = true
+                void scheduler
+                    .processEach(this._bufferedInvocations, (globals) => app.processEvent?.(globals))
+                    .then(() => {
+                        app.processedBuffer = true
+                        if (Object.values(this.apps).every((app) => app.processedBuffer || app.errored)) {
+                            this._stopBuffering?.()
```

Comment:
:warning: You're running this here and then also on line 106.  
  
Is this intended? One of them should probably go away IIUC